api/v1alpha1/wasm_types.go

```diff
@@ -78,8 +78,9 @@ type WasmCodeSource struct {
 	Image *ImageWasmCodeSource `json:"image,omitempty"`
 
 	// SHA256 checksum that will be used to verify the wasm code.
-	// +optional
-	// SHA256 *string `json:"sha256,omitempty"`
+	//
+	// kubebuilder:validation:Pattern=`^[a-f0-9]{64}$`
+	SHA256 string `json:"sha256"`
```

Comment:
Make SHA256 mandatory for now to avoid downloading wasm and calculate SHA in the control plane.

We can make it optional in the future without breaking changes if we opt to build a cache in the control plane.